module/merintr/alias.c

```diff
@@ -277,6 +277,8 @@ void AliasSave(void)
 	 }
       }
    }
+
+   AliasInit();
```

Comment:
I don't know what the underlying problem is, but this doesn't strike me as the right answer.  A function called "Save" should save, not load.

Whoever is calling AliasSave should also be adjusting the in-memory structures.  Can you figure out which caller isn't doing that correctly?

Reply:
Understood.

It looks like the in memory state was not saved when AliasSave was called to write to the .ini file, and that appears to be the cause of the issue.

I added in 2 in-memory saves when closing the `hotkey aliases` dialog box
1. either `interface_key_table` or `gCustomKeys` depending on whether the client is using classic keys or not
2. `inventory_key_table`

This is the same logic used when `AliasInit` is called.

From limited testing it now seems to work properly and the client does not require a restart to switch a hotkey alias from text command to text insert (or vice versa)